org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/plain/AddJarsCommand.java

```diff
@@ -0,0 +1,113 @@
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+/**
+ * Adds all jars from library folder into project's classpath.
+ *
+ * @author Valeriy Svydenko
+ */
+public class AddJarsCommand {
+  /**
+   * Adds jars into classpath.
+   *
+   * @param arguments contains two arguments: project uri and library folder's name
+   * @param pm a progress monitor
+   * @return uri of the project
+   */
+  public static Object execute(List<Object> arguments, IProgressMonitor pm) {
+    Preconditions.checkArgument(
+        arguments.size() >= 2, "Project uri and library folder are expected");
+
+    final String projectUri = (String) arguments.get(0);
+    final String libFolder = (String) arguments.get(1);
+
+    String projectName = projectUri.substring(projectUri.lastIndexOf(PATH_SEPARATOR) + 1);
+
+    IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
+    if (project == null || !project.exists()) {
+      throw new IllegalArgumentException(format("Project for '%s' not found", projectUri));
+    }
+
+    IJavaProject jProject = JavaModelUtil.getJavaProject(projectUri);
+    if (jProject == null) {
+      throw new IllegalArgumentException(format("Project for '%s' not found", projectUri));
+    }
+
+    IFolder lib = project.getFolder(libFolder);
+    if (!lib.exists()) {
+      throw new IllegalArgumentException(format("Folder for '%s' not found", libFolder));
+    }
+
+    List<IClasspathEntry> classpath = new ArrayList<>();
+    try {
+      classpath.addAll(findJars(jProject, lib));
+      classpath.addAll(Arrays.asList(jProject.getRawClasspath()));
+      jProject.setRawClasspath(
+          classpath.toArray(new IClasspathEntry[classpath.size()]),
+          jProject.getOutputLocation(),
+          pm);
+    } catch (CoreException e) {
+      JavaLanguageServerPlugin.log(e);
+    }
+    return projectUri;
+  }
+
+  private static List<IClasspathEntry> findJars(IJavaProject jProject, IFolder lib)
+      throws CoreException {
+    List<IClasspathEntry> jars = new ArrayList<>();
+    lib.accept(
+        proxy -> {
+          if (IResource.FILE != proxy.getType()) {
+            return true;
+          }
+
+          IPath path = proxy.requestFullPath();
+          if (!path.toString().endsWith(".jar")) {
+            return false;
+          }
+
+          IClasspathEntry libEntry =
+              newLibraryEntry(proxy.requestResource().getLocation(), null, null);
+
+          if (jProject.getClasspathEntryFor(libEntry.getPath()) == null) {
```

Comment:
It would be clearer if we returned all jars and did the check whether the jar is already on the classpath in the outer method. The performance impact is negligable, as we're talking about 5-10 items.